cpp/src/arrow/python/python_to_arrow.cc

```diff
@@ -79,7 +79,7 @@ class SequenceBuilder {
   Status AppendNone() {
     RETURN_NOT_OK(offsets_.Append(0));
     RETURN_NOT_OK(types_.Append(0));
-    return nones_.AppendToBitmap(false);
+    return nones_.AppendNull();
```

Comment:
@pcmoritz or @robertnishihara take a look to make sure these changes look right. 

Should this be `Append(false)`? I would expect `nones_` to be an all non-null boolean array